test/unit/compile-uri/compile-params-test.coffee

```diff
@@ -0,0 +1,199 @@
+{expect} = require 'chai'
+fury = new require 'fury'
+
+compileParams = require '../../../src/compile-uri/compile-params'
+
+describe 'compileParams', ->
+  it 'should compile a primitive href variable', ->
+    hrefVariables = new fury.minim.elements.HrefVariables()
+    hrefVariables.set('name', new fury.minim.elements.String())
+
+    parameters = compileParams(hrefVariables)
+
+    expect(parameters).to.deep.equal({
```

Comment:
We use `assert` everywhere in Dredd, but that's simple change and can be done later.